### PR TITLE
feat: create fetch_context

### PR DESCRIPTION
**Pull request by clairesquires**
https://trello.com/c/HOEisSkh/41-1-implementation-aadev-i-can-find-the-5-most-similar-embeddings-to-the-embedded-user-prompt

### CODE REVIEW

File: src/utils/fetchContext.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+);
+
+export const fetchContext = async (query_vector) => {
+  const response = await supabaseClient.rpc('vector_search', {
+    query_vector,
+    match_threshold: 0.78,
+    match_count: 5,
+    min_content_length: 0,
+  });
+  // TODO: error checking (response.error)
```
Comment (matt-newhall) [inline src/utils/fetchContext.ts:18]: todo? 👀

File: supabase/migrations/20230418100618_vector_search.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+create or replace function vector_search(query_vector vector(1536), match_threshold float, match_count int, min_content_length int)
+returns table (id bigint, input_text text, input_url text, usage_count bigint, similarity float)
+language plpgsql
+as $$
+#variable_conflict use_variable
```
Comment (matt-newhall) [inline supabase/migrations/20230418100618_vector_search.sql:5]: what does this mean?

File: supabase/migrations/20230418100618_vector_search.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+begin
+  return query
+  select
+    embedding.id,
+    embedding.input_text,
+    embedding.input_url,
+    embedding.usage_count,
+    (embedding.vector <#> query_vector) * -1 as similarity
```
Comment (matt-newhall) [inline supabase/migrations/20230418100618_vector_search.sql:13]: this and line 20 are showing as comments on git, do these work?

File: src/utils/fetchContext.ts (hunk 1 above)
Comment (clairesquires) [reply src/utils/fetchContext.ts:18]: Will write a ticket for doing this :)

File: supabase/migrations/20230418100618_vector_search.sql (hunk 2 above)
Comment (clairesquires) [reply supabase/migrations/20230418100618_vector_search.sql:5]: I don't know

File: supabase/migrations/20230418100618_vector_search.sql (hunk 3 above)
Comment (SimonLTheodo) [reply supabase/migrations/20230418100618_vector_search.sql:13]: They do! It's how the dot product is calculated

**Review by matt-newhall — APPROVED** [review]

**Pull request merged**